docs/api/metadata.md

```diff
@@ -0,0 +1,219 @@
+---
+id: metadata
+title: Extending Models Using Metadata
+---
+
+Saleor gives you the possibility to customize your shop without the need of any changes in the code. Once you get your own Saleor instance, you can extend most of Saleor's database models using metadata. 
+
+For example, you can add specific fields to metadata of objects to store extra information on that object. Using metadata you can also customize the way your storefront appears to your customers. 
+
+The following models contain metafields which data can be modified:
+
+* `collection`
+
+* `category`
+
+* `product`
+
+* `productVariant`
+
+* `productType`
+ 
+* `user`
+
+* `attribute`
+
+* `serviceAccount`
+
+* `checkout`
+
+Each metafield has four components creating the metadata:
+
+* `namespace` - Name to distinguish the metadata contained in this metafield from other metafields.
+
+* `clients` - It separates the `namespace` into subsections for specific `key` and `value` pairs as there can be multiple of them.
+
+* `key` - This is the name of your metafield.
+
+* `value` - This is the metafield's content.
+
+## Sample queries 
+
+The below examples show a sample query and response containing metadata:
+
+### Sample query with metadata
+
+```
+{
+  products(first:1){
+    edges{
+      node{
+        id
+        name
+        meta{
+          namespace
+          clients{
+            name
+            metadata{
+              key
+              value
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### Sample response from query with metadata
+
+```
+{
+  "data": {
+    "products": {
+      "edges": [
+        {
+          "node": {
+            "id": "UHJvZHVjdDo4Nw==",
+            "name": "Yellow Plimsolls",
+            "meta": [
+              {
+                "namespace": "taxes",
+                "clients": [
+                  {
+                    "name": "vatlayer",
+                    "metadata": [
+                      {
+                        "key": "code",
+                        "value": "standard"
+                      },
+                      {
+                        "key": "description",
+                        "value": ""
+                      }
+                    ]
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+## Making changes in metafields
+
+You can also add, update or remove metadata. 
+
+### Sample mutation to add or update metadata
+
+```
+mutation {
+  productUpdateMetadata(id:"UHJvZHVjdDo3Mg==", input:{namespace:"my_namespace1", clientName: "custom_name", key:"special_product", value:"true"}){
+    errors{
+      field
+      message
+    }
+    product{
+		meta{
```

Comment:
@korycins Please format the example queries properly. You can use "Prettify" option in the Playground. Not only this one but all of them, some of them have pretty long attributes lists.

Reply:
My bad! I've asked Maciek to paste the code snippets into the task in Jira, so I can place them in the docs. @korycins looks like you will have to redo them anyway :( sorry for the doubkle work!